drivers/spi/spi_nxp_s32.c

```diff
@@ -671,7 +671,7 @@ static const struct spi_driver_api spi_nxp_s32_driver_api = {
 			SPI_MCR_PCSIS(BIT_MASK(SPI_NXP_S32_NUM_CS(n))) |		\
 			SPI_MCR_MDIS(0U) | SPI_MCR_XSPI(1U) | SPI_MCR_HALT(1U)),	\
 		.TransferMode = SPI_IP_POLLING,						\
-		.StateIndex   = SPI_NXP_S32_HW_INSTANCE(n),				\
+		.StateIndex   = n,							\
```

Comment:
okay, but this is not related to the HAL version migration. It's an improvement done in this commit https://github.com/zephyrproject-rtos/zephyr/pull/65044/commits/e03a44976861cf1115d6f205e9dbadec8b3d2efe unrelated to the HAL version. So either move it to a separate pr, or update the commit message accordingly to not be misleading.

Reply:
yes, I update commit msg